### ISSUE TITLE
New Relic Insights

### ISSUE THREAD

**Issue by batmike** [ProxiBlue/NewRelic#12]
Insights is currently in Beta but I've been playing around with it and it's very powerful in terms of the general information that New Relic collects. It's also ridiculously fast and real-time, which is kind of fun to watch while page view and transaction numbers update every few seconds.

Anyway, what would be a very cool addition to this module would be the ability to define custom attributes to add in to various places in Magento that would then send off that data to New Relic. For example, the ability to insert the required attribute code into the checkout confirmation page to send off info to New Relic such as product name or product category or even things like payment method or shipment method.

You can see some info on custom attributes at https://docs.newrelic.com/docs/insights/custom-attributes and if you're interested or think it would be a useful or reasonable thing to add into this module, you can let me know and you can also take a look at our New Relic Insights beta account, if you want to check it out.

**Comment by barbazul** [comment]
Hi,

I got here searching for any custom attributes related issues. I don't care much for Insights atm but it would be great to start collecting some simple data like request parameters. That would greatly improve debugging slow traces like this one https://www.diigo.com/item/image/42au8/ymn1

What query parameters trigger the slow trace? I don't know.

It's a naive example but you get the idea.

There are some privacy concerns though that should be taken into consideration. If we start collecting logged in user id, email or even quote_id I have no idea what sort of privacy line we would be crossing.

If you could push your branch with whatever work you have put into this, I can put some hours into it as this is a feature I need asap.

**Comment by ProxiBlue** [comment]
Hi,

sadly I never managed to locate the work I did on the insights.
I have not been able to find any branch I had committed with this work.
I suspect I had worked on it, inside a clients site (as a branch) on a previous employment

I do still have a lot of repo's of old code to hunt through, so will attempt a new concerted effort to find it, but I have low hopes.

I recall I got the basic API code done. Frustrated to have potentially lost this code.
sorry.
At present I don't have any time free to continue active development on this module for additional features. Bugs fixes I will always have time for, as they are essential.
